src/core/geometry/qgstriangle.h

```diff
@@ -246,85 +311,134 @@ class CORE_EXPORT QgsTriangle : public QgsPolygonV2
      * \param lengthTolerance The tolerance to use
      * \returns The orthocenter of the triangle.
+     * An empty point is returned for empty or invalid triangle.
      * * Example:
      * \code{.py}
      *   tri = QgsTriangle( QgsPoint( 0, 0 ), QgsPoint( 0, 5 ), QgsPoint( 5, 5 ) )
      *   tri.orthocenter().asWkt()
      *   # 'Point (0 5)'
+     *   QgsTriangle().orthocenter().asWkt()
+     *   # 'Point (0 0)'
+     *   QgsTriangle( QgsPoint( 0, 0 ), QgsPoint( 5, 5 ), QgsPoint( 10, 10 ) ).orthocenter().asWkt()
+     *   # 'Point (0 0)'
+     *   QgsTriangle( QgsPoint( 0, 0 ), QgsPoint( 0, 0 ), QgsPoint( 10, 10 ) ).orthocenter().asWkt()
+     *   # 'Point (0 0)'
      * \endcode
      */
     QgsPoint orthocenter( double lengthTolerance = 0.0001 ) const;
 
     /**
      * Center of the circumscribed circle of the triangle.
-     * \returns The center of the circumscribed circle of the triangle
+     * \returns The center of the circumscribed circle of the triangle.
+     * An empty point is returned for empty or invalid triangle.
      * * Example:
      * \code{.py}
      *   tri = QgsTriangle( QgsPoint( 0, 0 ), QgsPoint( 0, 5 ), QgsPoint( 5, 5 ) )
      *   tri.circumscribedCenter().asWkt()
      *   # 'Point (2.5 2.5)'
+     *   QgsTriangle().circumscribedCenter().asWkt()
+     *   # 'Point (0 0)'
+     *   QgsTriangle( QgsPoint( 0, 0 ), QgsPoint( 5, 5 ), QgsPoint( 10, 10 ) ).circumscribedCenter().asWkt()
+     *   # 'Point (0 0)'
+     *   QgsTriangle( QgsPoint( 0, 0 ), QgsPoint( 0, 0 ), QgsPoint( 10, 10 ) ).circumscribedCenter().asWkt()
+     *   # 'Point (0 0)'
      * \endcode
      */
     QgsPoint circumscribedCenter() const;
 
     /**
      * Radius of the circumscribed circle of the triangle.
-     * \returns The radius of the circumscribed circle of the triangle
+     * \returns The radius of the circumscribed circle of the triangle.
+     * 0.0 is returned for empty or invalid triangle.
      * * Example:
      * \code{.py}
      *   tri = QgsTriangle( QgsPoint( 0, 0 ), QgsPoint( 0, 5 ), QgsPoint( 5, 5 ) )
      *   tri.circumscribedRadius()
      *   # 3.5355339059327378
+     *   QgsTriangle().circumscribedRadius()
+     *   # 0.0
+     *   QgsTriangle( QgsPoint( 0, 0 ), QgsPoint( 5, 5 ), QgsPoint( 10, 10 ) ).circumscribedRadius()
+     *   # 0.0
+     *   QgsTriangle( QgsPoint( 0, 0 ), QgsPoint( 0, 0 ), QgsPoint( 10, 10 ) ).circumscribedRadius()
+     *   # 0.0
      * \endcode
      */
     double circumscribedRadius() const;
 
     /**
-    * Circumscribed circle of the triangle.
-    * @return The circumbscribed of the triangle with a QgsCircle.
-    * Example:
-    * \code{.py}
-    *   tri = QgsTriangle( QgsPoint( 0, 0 ), QgsPoint( 0, 5 ), QgsPoint( 5, 5 ) )
-    *   tri.circumscribedCircle()
-    *   # QgsCircle(Point (2.5 2.5), 3.5355339059327378, 0)
-    * \endcode
-    */
+     * Circumscribed circle of the triangle.
+     * @return The circumbscribed of the triangle with a QgsCircle.
```

Comment:
I think it should be `\returns` instead of `@return`?

Reply:
done